### PR TITLE
feat(console): connector warnings in sign in methods

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Add connector warnings: 
1. When a sign in methods is enabled (primary/secondary), check if the corresponding connectors are enabled, and display a warning when needed.
2. If no social connector is added, show a warning.

## Linear Issue Reference

LOG-1741

## Testing

截屏2022-04-29 下午4 27 01

截屏2022-04-29 下午4 30 13

### CODE REVIEW

File: packages/console/src/pages/SignInExperience/components/ConnectorSetupWarning.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  type: ConnectorType;
+};
+
+const ConnectorSetupWarning = ({ type }: Props) => {
+  const { data: connectors } = useSWR<ConnectorDTO[], RequestError>('/api/connectors');
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+
+  if (
+    connectors &&
+    !connectors.some(({ metadata, enabled }) => metadata.type === type && enabled)
```
Comment (simeng-li) [inline packages/console/src/pages/SignInExperience/components/ConnectorSetupWarning.tsx:19]: will `!connectors?.some()` do ?

File: packages/console/src/pages/SignInExperience/components/SignInMethodsForm.tsx [hunk 2]
```diff
@@ -49,10 +52,19 @@ const SignInMethodsForm = () => {
...
             />
+            {method === SignInMethodKey.Email && email && (
+              <ConnectorSetupWarning type={ConnectorType.Email} />
+            )}
+            {method === SignInMethodKey.SMS && sms && (
+              <ConnectorSetupWarning type={ConnectorType.SMS} />
+            )}
+            {method === SignInMethodKey.Social && social && (
+              <ConnectorSetupWarning type={ConnectorType.Social} />
+            )}
```
Comment (simeng-li) [inline packages/console/src/pages/SignInExperience/components/SignInMethodsForm.tsx:63]: should this switch logic implement within the ConnectorSetupWarning itself?

File: packages/console/src/pages/SignInExperience/components/SignInMethodsForm.tsx [hunk 3]
```diff
@@ -74,6 +86,13 @@ const SignInMethodsForm = () => {
           )}
         />
       </FormField>
+      {primaryMethod === SignInMethodKey.SMS && <ConnectorSetupWarning type={ConnectorType.SMS} />}
+      {primaryMethod === SignInMethodKey.Email && (
+        <ConnectorSetupWarning type={ConnectorType.Email} />
+      )}
+      {primaryMethod === SignInMethodKey.Social && (
+        <ConnectorSetupWarning type={ConnectorType.Social} />
+      )}
```
Comment (simeng-li) [inline packages/console/src/pages/SignInExperience/components/SignInMethodsForm.tsx:95]: ditto

File: packages/console/src/pages/SignInExperience/components/ConnectorSetupWarning.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        action="admin_console.sign_in_exp.setup_warning.setup"
+        href={type === ConnectorType.Social ? '/connectors/social' : '/connectors'}
+      >
+        {t('sign_in_exp.setup_warning.no_connector', { context: type.toLowerCase() })}
+      </Alert>
+    );
+  }
+
+  return null;
+};
```
Comment (demonzoo) [inline packages/console/src/pages/SignInExperience/components/ConnectorSetupWarning.tsx:32]: const showWarning = connectors?.some(...);

return (
  showWarning && <Alert>...</Alert>
);

File: packages/console/src/pages/SignInExperience/components/SignInMethodsForm.tsx (hunk 2 above)
Comment (demonzoo) [reply packages/console/src/pages/SignInExperience/components/SignInMethodsForm.tsx:63]: makes sense

File: packages/console/src/pages/SignInExperience/components/ConnectorSetupWarning.tsx (hunk 4 above)
Comment (wangsijie) [reply packages/console/src/pages/SignInExperience/components/ConnectorSetupWarning.tsx]: It won't work, we can not return `false` in component.

File: packages/console/src/pages/SignInExperience/components/SignInMethodsForm.tsx (hunk 2 above)
Comment (wangsijie) [reply packages/console/src/pages/SignInExperience/components/SignInMethodsForm.tsx]: updated

File: packages/console/src/pages/SignInExperience/components/ConnectorSetupWarning.tsx (hunk 1 above)
Comment (wangsijie) [reply packages/console/src/pages/SignInExperience/components/ConnectorSetupWarning.tsx]: NO, they are not equal. But I make a small changes on this, PTAL.

**Review by simeng-li — APPROVED** [review]
LGTM

File: packages/phrases/src/locales/zh-cn.ts [hunk 5]
```diff
@@ -406,7 +406,15 @@ const translation = {
       },
       setup_warning: {
         setup: '设置',
-        no_connector: '你尚未添加社交登录连接器。你需要先完成设置才能启用。',
+        no_connector: '',
```
Comment (IceHe) [inline packages/phrases/src/locales/zh-cn.ts:409]: why remove the content of `no_connector`?

Comment (wangsijie) [reply packages/phrases/src/locales/zh-cn.ts:409]: Splitted into sub items.

**Pull request merged**